dmd-frontend/src/pages/login/index.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useCallback, useEffect } from 'react';
+import styled from 'styled-components';
+import { useForm } from 'react-hook-form';
+
+import Grid from '@material-ui/core/Grid';
+import Paper from '@material-ui/core/Paper';
+import Button from '@material-ui/core/Button';
+import Container from '@material-ui/core/Container';
+import TextField from '@material-ui/core/TextField';
+import Typography from '@material-ui/core/Typography';
+
+import {
+    checkPassword,
+    checkUsername,
+} from '../../utils/regex';
+
+const Login: React.FC<any> = (props: any) => {
+
+    const { handleSubmit, register, errors, setValue } = useForm<{
+        username: string,
+        password: string,
+    }>({
+        defaultValues: {
+            username: '',
+            password: '',
+        }
+    });
+
+    const _handleFormSubmit = useCallback((data, e) => {
+        e.preventDefault();
+        console.log(data);
+    }, []);
+
+    const _handleUsernameChange = useCallback((e) => {
+        setValue('username', e.target.value);
+    }, [setValue]);
+
+    const _handlePasswordChange = useCallback((e) => {
+        setValue('password', e.target.value);
+    }, [setValue]);
+
+    useEffect(() => {
+        register('username', {
+            validate: (value) => {
+                if (value.length === 0) return 'Username cannot be empty';
+                return checkUsername(value) || 'Invalid username format';
+            }
+        });
+        register('password', {
+            validate: (value: string) => {
+                if (value.length === 0) return 'Password cannot be empty';
+                if (value.length < 8 || value.length > 32) return 'Password must be between 8 and 32 characters';
+                return checkPassword(value) || 'Invalid password, please have a look at our password policy';
+            }
+        });
+    }, [register]);
+
+    return (
+        <FlexContainer
```

Comment:
Who the hell using this code style?

Reply:
@chiro-hiro , Which line exactly ?

Reply:
If it's about those that are in `useEffect` then I'm planning to move them out.

Reply:
@Marktine Line 58 and the rest of file :|.

Reply:
> ```
>                         <Grid
>                             container
>                             direction="column"
>                         >
> ```
> 
> Don't use this code style please, use this:
> 
> ```
>                         <Grid container direction="column">
> ```
Ohhh, Okay, I'll modify them